### PR TITLE
add query validator api for autodoc plugin

### PR DESCRIPTION
**Pull request by hanglics**

### CODE REVIEW

File: api.go [hunk 1]
```diff
@@ -138,31 +138,39 @@ func HandleQueryValidation(c *gin.Context) {
...
-			extractedFields = append(extractedFields, rawFields[0][1])
+			if len(rawFields) > 0 {
+				extractedFields = append(extractedFields, rawFields[0][1])
+			}
 		}
-		for _, i := range extractedFields {
-			flag := checker.CheckWord(fieldsDictionary, strings.ToLower(i), 0)
-			if !flag {
-				fieldsError = append(fieldsError, i)
+		if len(extractedFields) > 0 {
```
Comment (hscells) [inline api.go]: This check doesn't do anything, since Go will not range over an empty slice.

File: api.go [hunk 2]
```diff
@@ -138,31 +138,39 @@ func HandleQueryValidation(c *gin.Context) {
...
 		}
-	} else if lang == "pubmed" {
+	} else if strings.ToLower(lang) == "pubmed" {
 		reg := regexp.MustCompile(`\[([^]]+)\]`)
 		rawFields := reg.FindAllStringSubmatch(rawQuery, -1)
-		for _, i := range rawFields {
-			flag := checker.CheckWord(fieldsDictionary, strings.ToLower(i[1]), 0)
-			if !flag {
-				fieldsError = append(fieldsError, i[1])
+		if len(rawFields) > 0 {
```
Comment (hscells) [inline api.go]: Same thing with this check, can be removed (see above).

**Review by hscells — CHANGES_REQUESTED** [review]

File: api.go (hunk 1 above)
Comment (hanglics) [reply api.go]: Actually, the `extractedFields` can be `[]` if there is no fields presented in the query, like for Medline, `1 or 2` or for Pubmed, `brain and cancer`.

File: api.go (hunk 2 above)
Comment (hanglics) [reply api.go]: Same as above.

**Review by hscells — APPROVED** [review]

**Review by hscells — APPROVED** [review]

**Pull request merged**